test/common/verifier.go

```diff
@@ -0,0 +1,144 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package common
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/go-logr/logr"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// Verifier is a struct that can be used to verify whether the shoot-rsyslog-relp extension is working as expected.
+type Verifier struct {
+	log                        logr.Logger
+	client                     kubernetes.Interface
+	rsyslogRelpEchoServerPodIf clientcorev1.PodInterface
```

Comment:
It should be possible to obtain the PodInterface from the `client`: `client.Kubernetes().CoreV1().Pods(...)`.

Reply:
If I use the echo server deployed in the kind cluster for the e2e tests, I still need this interface since the verifier requires a client to the shoot to verify that rsyslog is configured on the nodes and to generate logs, and then it needs a pod interface to the `rsyslog-relp-echo-server` pod running in the kind cluster.